### PR TITLE
Backport of the "jdt" block from the old eclipse.gradle file

### PR DESCRIPTION
**Pull request by zosrothko**
### Context
[issue #5830](https://github.com/gradle/gradle/issues/5830)

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt [hunk 1]
```diff
@@ -88,6 +88,20 @@ open class IdePlugin : Plugin<Project> {
                         }
                     })
                 }
+                jdt {
+                    sourceCompatibility = JavaVersion.VERSION_1_8
+                    targetCompatibility = JavaVersion.VERSION_1_8
+                    javaRuntimeName = "JavaSE-1.8"
+/*
+FIXME: No idea why Kotlin does not compile the lines below
```
Comment (marcphilipp) [inline buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt]: What's the error message?

Comment (zosrothko) [reply buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt]: with the code below
```
jdt {
     sourceCompatibility = JavaVersion.VERSION_1_8
     targetCompatibility = JavaVersion.VERSION_1_8
     javaRuntimeName = "JavaSE-1.8"
				
     file {
          withProperties { Properties properties ->        // <== line 97
         // Eclipse's view of projects treat circular dependencies as errors by default
         properties["org.eclipse.jdt.core.circularClasspath"] = "warning"
     }
}
```
Kotlin states that
```
C:\Users\fandre\Documents\git\gradle>gradlew build
Parallel execution is an incubating feature.

> Task :buildSrc:ide:ktlintMainCheck FAILED
C:\Users\fandre\Documents\git\gradle\buildSrc\subprojects\ide\src\main\kotlin\org\gradle\gradlebuild\ide\IdePlugin.kt:97:63: Not a valid Kotlin file (expecting an element)
```

Comment (zosrothko) [reply buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt]: Signed-off

Comment (marcphilipp) [reply buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt]: Looks like there are 4 commits now and you only signed off one of them. Please rebase on `master` and sign both your original commits.

**Pull request closed without merging**